### ISSUE TITLE
Add python.app to the test matrix

### ISSUE THREAD

**Issue by jni** [napari/napari#1598]
## 🐛 Bug

@joshmoore pointed out in #1597 that our command line interface was failing when used with `pythonw`/`python.app`. We should add tests with and without python.app in our testing matrix to prevent a regression.

**Comment by sofroniewn** [comment]
Agreed, would be nice to add this

**Comment by sofroniewn** [comment]
@tlambert03 should this be added to #2051 too? Maybe as it's own thing? I'm a little wary of test explosion, but curious what you think

**Comment by tlambert03** [comment]
I can do that, let's address #2051 as is though, and it will be easier to play around with and merge afterwards

**Comment by sofroniewn** [comment]
Waiting till after #2051 sounds good, no rush on this.

**Comment by jaimergp** [comment]
I took a look at this and how to potentially implement it. Since we are using tox, the `tox-conda` plugin looked like the obvious choice, but right now it's either all-in or nothing - you can't select which tox entries use conda. If this issue (https://github.com/tox-dev/tox-conda/issues/36) gets solved, then we could add a new entry to `tox.ini`.

The other option is to turn all tox envs into conda envs with `tox-conda` (deps still installed from PyPI), but that might be too much of a change (and potentially slower?) for the CI.

We can also implement it without `tox`, adding some duplication. What would the preferred approach?

* Wait for tox-conda to be more granular
* Use tox-conda for all envs
* Implement outside the tox workflow

**Comment by jni** [comment]
I think options 1 and 2 would be fine, outside the tox workflow would add a bit too much complexity I think.

**Comment by willingc** [comment]
Next step: verify need and priority of this test issue.

**Comment by psobolewskiPhD** [comment]
Is this still a thing? i tried to find info about python.app but struggled. i think it was a way to get gui python things to work on macOS? Maybe only older macOS? It's no longer listed on the python docs:
https://docs.python.org/3/using/mac.html#gui-programming
And it's not needed for napari currently on modern-ish macOS as far as I can tell.

Any macOS python old-timers care to chime in? can we just close this?

**Comment by willingc** [comment]
@psobolewskiPhD I would go ahead and close. I believe it was when Python distributed support for OS X. The last known reference that I could find was 3.6 and that is already way past end of life.
